pkg/cmd/run.go

```diff
@@ -93,8 +115,11 @@ func RunMicroshift(cfg *config.Config) error {
 		klog.Fatalf("MicroShift must be run privileged")
 	}
 
+	cleanUpPreviousFileLogs()
+
 	if err := performPrerun(); err != nil {
 		klog.ErrorS(err, "Pre-run procedure failed")
+		writeFileLogError(preRunFailedLogPath, err)
```

Comment:
This is good. I was thinking we might record the error from `RunMicroshift` in case something after the pre-run phase triggers the error, but let's start with this and see if it's useful to expand the scope later.

Reply:
Sure, that's a trivial change if need be, since we were calling out pre-run specifically in this ticket that's the reason it's still oriented around that, but we can change it up to capture all start up errors and propagate that up through greenboot.

Reply:
Prerun is now split into two phases: data management and version checking/updating which happens in line 147 (`prerun.CheckAndUpdateDataVersion()`) which we also need to catch.

We might need to rename some functions to improve readability, but for now:
if we want to persist only prerun errors, we might wrap errors from these two functions and only persist if that's the error type, something like in RunE func:
```
if err := RunMicroShift(...); err != nil {
    if errors.Is(err, FailedPreRun) {
        writeFileLogError(...)
    }
}
```

Reply:
I'm thinking we avoid custom type errors for now since that would be a bigger conversation. I think just capturing both `performPrerun` and `prerun. CheckAndUpdateDataVersion()` would be good enough?

Reply:
From slack conversation, keeping this simple and focusing only on pre-run errors.